lib/Jsrt/ChakraCommon.h

```diff
@@ -2978,4 +2978,343 @@ typedef unsigned char* ChakraBytePtr;
+
+    /// <summary>
+    ///     TTD API -- may change in future versions:
+    ///     Construct a HANDLE that will be used to read/write the event log portion of the TTD data based on the uri 
+    ///     provided by JsTTDInitializeUriCallback.
+    /// </summary>
+    /// <remarks><para>Exactly one of read or write will be set to true.</para></remarks>
+    /// <param name="uri">The fully resolved location for the TTD data as provied by JsTTDInitializeUriCallback.</param>
+    /// <param name="read">If the handle should be opened for reading.</param>
+    /// <param name="write">If the handle should be opened for writing.</param>
+    /// <returns>A HANDLE opened in read/write mode as specified.</returns>
+    typedef HANDLE (CALLBACK *JsTTDGetLogStreamCallback)(_In_z_ const wchar_t* uri, _In_ bool read, _In_ bool write);
+
+    /// <summary>
+    ///     TTD API -- may change in future versions:
+    ///     Construct a HANDLE that will be used to read/write a snapshot and generate a unique uri that is associated with this snapshot.
+    /// </summary>
+    /// <remarks><para>Exactly one of read or write will be set to true.</para></remarks>
+    /// <param name="uri">The fully resolved root location for the TTD data as provied by JsTTDInitializeUriCallback.</param>
+    /// <param name="snapId">A unique string identifier for this snapshot.</param>
+    /// <param name="read">If the handle should be opened for reading.</param>
+    /// <param name="write">If the handle should be opened for writing.</param>
+    /// <returns>A HANDLE opened in read/write mode as specified.</returns>
+    typedef HANDLE (CALLBACK *JsTTDGetSnapshotStreamCallback)(_In_z_ const wchar_t* uri, _In_z_ const wchar_t* snapId, _In_ bool read, _In_ bool write);
+
+    /// <summary>
+    ///     TTD API -- may change in future versions:
+    ///     Construct a HANDLE that will be used to read/write information on source code loaded by the program.
+    /// </summary>
+    /// <remarks><para>Exactly one of read or write will be set to true.</para></remarks>
+    /// <param name="uri">The fully resolved root location for the TTD source code data.</param>
+    /// <param name="bodyCtrId">A unique string identifier for this source file.</param>
+    /// <param name="srcFileName">The base filename for this source code.</param>
+    /// <param name="read">If the handle should be opened for reading.</param>
+    /// <param name="write">If the handle should be opened for writing.</param>
+    /// <returns>A HANDLE opened in read/write mode as specified.</returns>
+    typedef HANDLE (CALLBACK *JsTTDGetSrcCodeStreamCallback)(_In_z_ const wchar_t* uri, _In_z_ const wchar_t* bodyCtrId, _In_z_ const wchar_t* srcFileName, _In_ bool read, _In_ bool write);
+
+    /// <summary>
+    ///     TTD API -- may change in future versions:
+    ///     A callback for reading data from a handle.
+    /// </summary>
+    /// <param name="strm">The HANDLE to read the data from.</param>
```

Comment:
Can we change it to - handle, dataHandle, fileHandle or fileStream. misspelled words are not intuitive.